### ISSUE TITLE
SCSS / SASS support

### ISSUE THREAD

**Issue by florbraz** [emmetio/livestyle-sublime-old#11]
##

**Comment by danhere** [comment]
Came here to say this as well. Sass support would make this insanely useful for me. Though I imagine it would be tougher to implement.

**Comment by denisnazarov** [comment]
Couldn't it be based on the way Chrome DevTools does Sass integration in its experiment now, using source maps? 

http://bricss.net/post/33788072565/using-sass-source-maps-in-webkit-inspector
http://www.html5rocks.com/en/tutorials/developertools/revolutions2013/#toc-sass-debugging

**Comment by sergeche** [comment]
No. Source maps are good for viewing things, but not for editing. Whenever you change something in SASS or generated CSS, source map become invalid and have to be re-generated again.

**Comment by maximilianschmid** [comment]
SASS Support (SCSS) would make this Tool useful to us.

**Comment by lmartins** [comment]
Not sure if this is the right place to say it, but I also add my vote on Sass support.
Thanks.

**Comment by andheiberg** [comment]
+1

**Comment by simonexmachina** [comment]
++1 :)

**Comment by romanonthego** [comment]
As much as I love the idea and appreciate work was done, I obligated to say that without SASS/LESS/Stylus support it simple can not be integrated into real work process (as rails/frontend developer I haven't seen vanila-CSS based project for years - even the small public projects is worth using some kind of pipeline) and tend to be more like fun and impressive experiment - but not the real tool. I simply can not afford to trade off the power and maintainability of preprocessors on any kind of not-a-contact-page projects today.

So:  +1 for kickstarter project, as soon as it brings SASS support (my vote is for Stylus actually, but SASS seems more real :)

**Comment by pontoffeltier** [comment]
+1 for Sass/Scss

**Comment by stevenbriscoeca** [comment]
+1! :)

**Comment by SahilMepani** [comment]
+1

**Comment by jamescostian** [comment]
Stylus would be great too!

**Comment by marciobda** [comment]
+1 . With Sass/Scss support, this will be insanely useful in my everyday projects.

**Comment by shuuheyhey** [comment]
+1 for Sass/Scss!

**Comment by ChrisC** [comment]
+1

**Comment by DaveOrDead** [comment]
+1

**Comment by jb510** [comment]
Not sure any more +1's are needed as obviously this would be awesome and is highly desired.

Is any work being done on implementing SCSS/SASS though?  

I don't know enough about web inspector and suspect my input is trivial, like telling fire gods how to rub sticks together.  But it seems like maybe you could overlay a user identified SASS file over the CSS in inspector, or ass a new tab and then recompile the SASS to CSS in javascript (I'm assuming there is a JS based SASS compiler, I've only ever used one for LESS).

Even without preprocessor support this is pretty awesome though.  Thanks.

**Comment by sergeche** [comment]
Re-compilation of SASS doesn’t makes sense for LiveStyle. It will be yet another LiveReload/CodeKit/etc. 

The goal of LiveStyle is to make a truly live (without any re-compilation) and bi-directional development. I’m finishing _Milestone 1_ release of LiveStyle right now and the next _Milestone 2_ will be focused on enabling and improving LESS and SCSS experimental support. At least you can use it for basic bi-directional live tweaking together with tools like `grunt-livereload`, e.g. make simple tweaks with LiveStyle and make full re-compilation/page reload on file saving.

**Comment by lmartins** [comment]
Live editing with Sass makes a lot of sense, im using http://usetakana.com/ for that.
But I see your point about the bi-directional thing being anything but trivial.

**Comment by jb510** [comment]
Thanks Sergey.  The bi-directional aspect of LiveStyle is what I find compelling so I look forward to it.  I've been a little confused over SCSS/SASS/LESS support.  Does LS currently support SCSS? I know it was in a demo somewhere, but I wasn't clear if that was real or there were limits to what it could do.  Maybe it's obvious and I just haven't tried, but it seems totally undocumented.

**Comment by sergeche** [comment]
SCSS is just an experiment I made for demo video to show the power of LiveStyle engine. It is disabled because it wasn’t tested enough and may actually break your code. It will be enabled and improved in Milestone 2

**Comment by RaphaelDDL** [comment]
I'd like to +1 scss support.
I also understand why you can't add recompilation to LS (out of tool focus) nor can't use sourcemaps so will not expect so soon for sass projects. On other hand, is awesome for plain css ones!

**Comment by sergeche** [comment]
1. Re-compilation is not an option: it’s very slow. With re-compilation it would be just another CodeKit/LiveReload clone.
2. Source maps must be re-compilated for each update, which it also slow. And they don’t work in CSS→SCSS direction. For example, when you add new `padding` property in DevTools, it’s source does not exists in SCSS so it can’t be mapped at all.

**Comment by RaphaelDDL** [comment]
Yeah, probably anything related to sass would have to rely on re-compiling/sass/compass stuff. 

Maybe the way to make LS work would to code something to act as a plugin for compass for e.g., where it's `watch` would act bi-directional - that is - instead of:
- only listening to .scss
- comparing with .sass-cache
- recompiling if different

Would be:
- listen both .scss and .css
- comparing with .sass-cache
- recompile if .scss differs, apply changes to .scss if .css differs

This way LiveStyle would only need to keep working as is, editing final `.css`.

Don't know if sass/compass can predict the changes based on last sass-cache and do inverse compilation.. 
We gotta ask [nex3](http://github.com/nex3/) and [chriseppstein](http://github.com/chriseppstein/) for that.

**Comment by iaminawe** [comment]
+1

**Comment by Miguelos** [comment]
+1

**Comment by mcabrams** [comment]
+1

**Comment by Ulysses-Odysseus** [comment]
1 mas! Would love to have this!

**Comment by morekid** [comment]
+1

**Comment by byrichardpowell** [comment]
+1

**Comment by kaiye** [comment]
Waiting for LS Milestone 2, diff/re-compiling is not so hard, but performance is a big problem.

**Comment by simonexmachina** [comment]
FYI, performance is much better if you're using libsass rather than the
Ruby-based compiler.

**Comment by jb510** [comment]
@sergeche Any update on this?  I'm not seeing a dev branch or anything for this.  Just curious if SCSS/SASS support is being worked on or still just "some time in the future"?

**Comment by sergeche** [comment]
Working on it. LiveStyle is based entirely on Emmet so I have to finish v1.1: https://github.com/emmetio/emmet/issues?milestone=1

**Comment by jb510** [comment]
Great!  TY for the update!

**Comment by thecotne** [comment]
+1

**Comment by zeroxme** [comment]
+1

**Comment by beyondghx** [comment]
Nice work!Waiting for SASS~

**Comment by ldexterldesign** [comment]
+1

**Comment by franciscolourenco** [comment]
+1

**Comment by jcpplus** [comment]
+1

**Comment by wesstong** [comment]
+1

**Comment by theRemix** [comment]
+1

**Comment by hamsterbacke23** [comment]
+1

**Comment by lackneets** [comment]
+1

**Comment by OutThisLife** [comment]
+1

**Comment by Rad1calDreamer** [comment]
+1

**Comment by enapupe** [comment]
I still can't see how that is going to work.
Unless you make chrome understand scss, and this way, the plugin wouldn't even need a update..
How you gonna make it bi-directional? Source maps allows you to know the original SASS (less,etc) file   resource you need to edit, but how to to the opposite direction? Recompile/remap every save sounds not possible.

**Comment by maximkoretskiy** [comment]
+1

**Comment by PixelT** [comment]
+1

**Comment by bruncun** [comment]
+1

**Comment by zontafil** [comment]
+1!

**Comment by lucasvst** [comment]
+1

**Comment by gebeer** [comment]
+1

**Comment by mkormendy** [comment]
So where are we at with addressing this issue? There's an obvious interest in it but the author's last comment on it was over a year ago. Is anyone willing to help work on this or have any solutions?

**Comment by sergeche** [comment]
I’ll publish new alpha with LESS and SCSS support soon

**Comment by ldexterldesign** [comment]
:+1:

**Comment by sergeche** [comment]
Hey guys, thanks for waiting for so long :)

After a year of heavy development I’m glad to announce LiveStyle 0.9 Alpha with SCSS (not SASS) support.
Check it out: http://livestyle.emmet.io/alpha/

**Comment by mkormendy** [comment]
This is great news Sergey! I am going over the SCSS implementation as we speak to see what the limitations are. This is definitely a massive step of progress.

**Comment by Ulysses-Odysseus** [comment]
This are great news!! Awesome thank you!!

**Comment by astralscott** [comment]
Any chance SASS support is going to end up in here also?

**Comment by mkormendy** [comment]
If you're using SASS, you're steps away from SCSS, why not just use SCSS? A processor is still required and SCSS is more expressive, and does almost all of the things that SASS does and more. http://thesassway.com/editorial/sass-vs-scss-which-syntax-is-better

**Comment by sergeche** [comment]
SASS syntax has a low priority right now. I’m planning a crowdfunding campaign soon to support future LiveStyle development and cool features so SASS support will depend on user support :)
